api/client/cp.go

```diff
@@ -191,10 +193,29 @@ func (cli *DockerCli) copyFromContainer(srcContainer, srcPath, dstPath string, c
 		_, srcBase := archive.SplitPathDirEntry(srcInfo.Path)
 		preArchive = archive.RebaseArchiveEntries(content, srcBase, srcInfo.RebaseName)
 	}
+
+	var reader io.ReadCloser
+	if cli.isTerminalOut {
+		size := stat.Size
+		if stat.Mode.IsDir() {
+			size = stat.DSize
+		}
+		sf := streamformatter.NewStreamFormatter()
+		progressOutput := sf.NewProgressOutput(cli.out, true)
+		reader = progress.NewProgressReader(preArchive, progressOutput, size, "", "Copying")
+		defer func() {
+			// Format blank line
+			cli.out.Write(sf.FormatStatus("", ""))
```

Comment:
Great!! Thanx @aaronlehmann , I don't like the ugly "def func" way in my patch too. I will investigate why I did not get what I want with NewLines == true. Anyway, I would take the use of the infrastructure you mentioned above. :)